addons/controller_icons/ControllerIcons.gd

```diff
@@ -240,24 +250,29 @@ func get_path_type(path: String) -> PathType:
 	else:
 		return PathType.SPECIFIC_PATH
 
-func get_matching_event(path: String, input_type: InputType = _last_input_type) -> InputEvent:
+func get_matching_event(path: String, input_type: InputType = _last_input_type, controller: int = _last_controller) -> InputEvent:
 	var events : Array
 	if _custom_input_actions.has(path):
 		events = _custom_input_actions[path]
 	else:
 		events = InputMap.action_get_events(path)
 
+	var fallback = null
 	for event in events:
 		match event.get_class():
 			"InputEventKey", "InputEventMouse", "InputEventMouseMotion", "InputEventMouseButton":
 				if input_type == InputType.KEYBOARD_MOUSE:
 					return event
 			"InputEventJoypadButton", "InputEventJoypadMotion":
 				if input_type == InputType.CONTROLLER:
-					return event
-	return null
+					if event.device == controller:
+						return event
+					elif fallback == null:
+						fallback = event
+
+	return fallback
```

Comment:
Functionally, this has the same behavior from the previous code; if the `if` is `true`, return `event`; else, set `fallback` to `event`, and return `fallback`, which is `event`. Why was this needed?

Reply:
It is the same behavior only if all mappings use the "all devices" option. In case of device specific mappings, this makes it pick the correct mapping for the passed/most recently used controller. This is useful for stuff like local multiplayer or giving some controller types a different layout (like the mentioned A/B swap for Nintendo).

I was thinking about adding a `event.device == -1` check so it can only fall back to "all devices" mappings. I decided to use the current behavior as fallback, but adding that might actually be better (especially in the local multiplayer scenario with per player/controller mappings: for example some fighting games offer an optional shortcut instead of pressing multiple buttons simultaneously). What do you think?

Reply:
Ah yeah, the behavior is different since this is running on a for loop. I didn't notice it before, my bad.

> I was thinking about adding a event.device == -1 check so it can only fall back to "all devices" mappings. 

Do you mean adding this as an additional check, or replacing the current one? Either way, I trust your judgement whether to add it or not; since it seems to be relevant for the future when implementing #58, it will be easier to test its behavior once that is implemented.

Reply:
I added it as an additional check so it takes the first fallback mapping, just like it does for the device specific mappings.

Showing all matching mappings would be a cool feature but it might take a few weeks until I find time to implement it (needs a bigger refactoring of my game).